### PR TITLE
Updated the cropping algorithm for MoveNet models.

### PR DESCRIPTION
**Pull request by yuhuichen1015**
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/661)

### CODE REVIEW

**Review by ardoerlemans — APPROVED** [review]
Thanks for updating this!

File: pose-detection/src/movenet/detector.ts [hunk 1]
```diff
@@ -443,23 +409,75 @@ export class MoveNetDetector extends BasePoseDetector {
...
-        width: (cropCorner[1] + cropLength) / imageWidth -
-            cropCorner[1] / imageWidth
-      };
+  /**
+   * Provides initial crop region.
+   *
+   * The function provides the initial crop region when the algorithm cannot
+   * reliably determine the crop region from the previous frame. There are two
+   * scenarios:
+   *   1) The very frist frame: the function returns the best quess by cropping
```
[the remark this answers is not in the document]
Comment (yuhuichen1015) [reply pose-detection/src/movenet/detector.ts]: Thanks for the catch! Fixed.

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/661)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ardoerlemans and @yuhuichen1015)

___
*[pose-detection/src/movenet/detector.ts, line 219 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpL1H96RRmdX3PB6LU:-MYpL1H96RRmdX3PB6LV:bee5hfb) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L219)):*
> ```XML
> 
>     // After the first run, set the isFirstFrame to false.
>     this.isFirstFrame = false;
> ```

It seems you always want to set some cropRegion when cropRegion is null, this happens when it is first run or when no pose is detected in the last frame. Then why not just use `cropRegion == null` as the flag to trigger initCropRegion?
___
*[pose-detection/src/movenet/detector.ts, line 266 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpJ0NSBtCI97LC_kud:-MYpJ0NSBtCI97LC_kue:b7zujad) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L266)):*
> ```XML
>     } else {
>       // No pose detected, so reset all filters.
>       this.resetFilters();
> ```

Should cropRegion be reset too?
___
*[pose-detection/src/movenet/detector.ts, line 448 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpJUcS1EG7rkwcFO27:-MYpJUcS1EG7rkwcFO28:bmr4sjz) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L448)):*
> ```XML
>    *      image.
>    */
>   initCropRegion(imageHeight: number, imageWidth: number) {
> ```

nit: `private initCropRegion(`
___
*[pose-detection/src/movenet/detector.ts, line 451 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpKAEK2l3asKD4sktj:-MYpKAEK2l3asKD4sktk:b-qnqrhx) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L451)):*
> ```XML
>     let boxHeight, boxWidth, yMin, xMin;
>     if (this.isFirstFrame) {
>       // If it is the first frame, perform a best guess by making the square
> ```

Curious why for first frame, not run the model on the full image with padding on both sides/

**Review by yuhuichen1015 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/661)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128)

___
*[pose-detection/src/movenet/detector.ts, line 219 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpL1H96RRmdX3PB6LU:-MYpbIPvD4i-ezz4Sec-:b-ipf0f3) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L219)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

It seems you always want to set some cropRegion when cropRegion is null, this happens when it is first run or when no pose is detected in the last frame. Then why not just use `cropRegion == null` as the flag to trigger initCropRegion?
</blockquote></details>

That's actually a good point. Thanks for the suggestion. Removed the isFirstFrame flag and used the cropRegion instead.
___
*[pose-detection/src/movenet/detector.ts, line 266 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpJ0NSBtCI97LC_kud:-MYpbUoGBX1Xi7Rgy2xz:ba2by8k) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L266)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Should cropRegion be reset too?
</blockquote></details>

We actually try to only resetting the temporal filters but not the cropRegion. The initCropRegion should handle the case where there is no detected keypoints from the previous frame.
___
*[pose-detection/src/movenet/detector.ts, line 448 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpJUcS1EG7rkwcFO27:-MYpbj_DCanHPyYaqxd6:boyexbg) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L448)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

nit: `private initCropRegion(`
</blockquote></details>

Done. Thanks.
___
*[pose-detection/src/movenet/detector.ts, line 451 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpKAEK2l3asKD4sktj:-MYpbkfK-uSkrGT6YmPZ:bjc7kzh) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L451)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Curious why for first frame, not run the model on the full image with padding on both sides/
</blockquote></details>

Running the model on the full image with padding does not utilize the detection power efficiently, sometimes prevent the model from entering the cropping mode. Provide a good initial guess increases the chance of having good predictions from the beginning, resulting in good overall performance. Added some comment to describe in more detail.

**Review by lina128 — APPROVED** [review]
Thank you Yu-hui!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-:-MYpk-WJ7etw9hicjlfe:b-7werl2)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @lina128 and @yuhuichen1015)

___
*[pose-detection/src/movenet/detector.ts, line 451 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/661#-MYpKAEK2l3asKD4sktj:-MYpjaTu2KMoyNwuUj6Y:bkk6gst) ([raw file](https://github.com/tensorflow/tfjs-models/blob/5b18ad1459331cd405a20c40e629c1e1f0ab5add/pose-detection/src/movenet/detector.ts#L451)):*
<details><summary><i>Previously, yuhuichen1015 (Yu-Hui Chen) wrote…</i></summary><blockquote>

Running the model on the full image with padding does not utilize the detection power efficiently, sometimes prevent the model from entering the cropping mode. Provide a good initial guess increases the chance of having good predictions from the beginning, resulting in good overall performance. Added some comment to describe in more detail.
</blockquote></details>

Got it, make sense!

**Pull request merged**